lib/posix/CMakeLists.txt

```diff
@@ -19,26 +19,26 @@ if(CONFIG_POSIX_API OR CONFIG_PTHREAD_IPC OR CONFIG_POSIX_CLOCK OR
 endif()
 
 zephyr_library()
-zephyr_library_sources_ifdef(CONFIG_POSIX_API perror.c)
-zephyr_library_sources_ifdef(CONFIG_PTHREAD_IPC pthread_common.c)
-zephyr_library_sources_ifdef(CONFIG_PTHREAD_IPC pthread_cond.c)
-zephyr_library_sources_ifdef(CONFIG_PTHREAD_IPC pthread_mutex.c)
-zephyr_library_sources_ifdef(CONFIG_PTHREAD_IPC pthread_barrier.c)
-zephyr_library_sources_ifdef(CONFIG_PTHREAD_IPC pthread.c)
-zephyr_library_sources_ifdef(CONFIG_PTHREAD_IPC pthread_sched.c)
-zephyr_library_sources_ifdef(CONFIG_PTHREAD_IPC pthread_spinlock.c)
+add_subdirectory_ifdef(CONFIG_GETOPT getopt)
+zephyr_library_sources_ifdef(CONFIG_PTHREAD_IPC _common.c)
```

Comment:
Alternatively, sort by Kconfig alphabetically then files within that Kconfig group:
```
zephyr_library_sources_ifdef(CONFIG_EVENTFD eventfd.c)
zephyr_library_sources_ifdef(CONFIG_FNMATCH fnmatch.c)
zephyr_library_sources_ifdef(CONFIG_POSIX_API perror.c)
zephyr_library_sources_ifdef(CONFIG_POSIX_CLOCK clock.c)
zephyr_library_sources_ifdef(CONFIG_POSIX_CLOCK nanosleep.c)
zephyr_library_sources_ifdef(CONFIG_POSIX_CLOCK sleep.c)
zephyr_library_sources_ifdef(CONFIG_POSIX_CLOCK timer.c)
zephyr_library_sources_ifdef(CONFIG_POSIX_FS fs.c)
zephyr_library_sources_ifdef(CONFIG_POSIX_MQUEUE mqueue.c)
zephyr_library_sources_ifdef(CONFIG_PTHREAD_IPC _common.c)
zephyr_library_sources_ifdef(CONFIG_PTHREAD_IPC barrier.c)
zephyr_library_sources_ifdef(CONFIG_PTHREAD_IPC cond.c)
zephyr_library_sources_ifdef(CONFIG_PTHREAD_IPC key.c)
zephyr_library_sources_ifdef(CONFIG_PTHREAD_IPC mutex.c)
zephyr_library_sources_ifdef(CONFIG_PTHREAD_IPC pthread.c)
zephyr_library_sources_ifdef(CONFIG_PTHREAD_IPC rwlock.c)
zephyr_library_sources_ifdef(CONFIG_PTHREAD_IPC sched.c)
zephyr_library_sources_ifdef(CONFIG_PTHREAD_IPC semaphore.c)
zephyr_library_sources_ifdef(CONFIG_PTHREAD_IPC spinlock.c)
```

Reply:
Yeah, that makes sense too, I guess

Reply:
Fixed!